test/views/components/ErrorSpec.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.components
+
+import org.scalatest.{Matchers, WordSpec}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.inject.Injector
+import play.api.i18n.{Messages, MessagesApi, Lang}
+import play.api.test.FakeRequest
+import play.twirl.api.Html
+import org.jsoup.Jsoup
+import views.html._
+import models.ui.{Paragraph, Text, Question, Answer, ErrorMsg}
+import org.jsoup.nodes.{Document, Element}
+import scala.collection.JavaConverters._
+
+class ErrorSpec extends WordSpec with Matchers with GuiceOneAppPerSuite {
+
+  def asDocument(html: Html): Document = Jsoup.parse(html.toString)
+
+  trait Test {
+    private def injector: Injector = app.injector
+    def messagesApi: MessagesApi = injector.instanceOf[MessagesApi]
+    implicit def messages: Messages = messagesApi.preferred(Seq(Lang("en")))
+    val headStrings = Vector("HEADING", "Welsh, HEADING")
+    val errorStrings = Vector("Please select", "Welsh, Please")
+    val fakeRequest = FakeRequest("GET", "/")
+    val heading = Text(headStrings)
+    val errorMsgs = Seq(ErrorMsg("id", Text(errorStrings)))
+  }
+
+  "error_summary" must {
+
+    "render error header and list of messages" in new Test {
+      val doc = asDocument(components.error_summary(heading, errorMsgs)(messages))
+      val head = doc.getElementById("error-summary-title")
+
+      head.text() shouldBe headStrings(0)
+
+      val lists = doc.getElementsByTag("ul").asScala.toList
+      val listItems = lists(0).getElementsByTag("li").asScala.toList
+
+      listItems(0).text() shouldBe errorStrings(0)
+    }
+
+  }
+
+  "error_message" must {
```

Comment:
You could add another test here for the visually hidden text. You would probably have to identify the element by its class.